feat: add methods and classes to execute low level HTTP requests in the ksqldb-api-client (#8118)
confluentinc/ksql: docs/developer-guide/ksqldb-clients/java-client.md

Map<String, Object> variables = client.getVariables();
```

Execute Direct HTTP Requests<a name="direct-http-requests"></a>
---------------------------------------------------------------

> Sometimes it is necessary to execute requests directly against the ksqlDB server for reasons including, 

Reviewer: ```suggestion
Sometimes, you need to execute requests directly against the ksqlDB server for reasons including, 
```

but not limited to, accessing features in ksqlDB REST API that are not available in the API client,
or deserializing responses into different classes that are more native to your application.

> For this, the Client now adds an `HttpRequest` and `HttpResponse` interface that can be used to send 

Reviewer: ```suggestion
For this purpose, the Client now adds an `HttpRequest` and `HttpResponse` interface that you can use to send 
```

direct requests. 

### Example Usage ###
Call the `/info` endpoint in ksqlDB with: 
```java
HttpResponse response = client.buildRequest("GET", "/info")
    .send()
    .get();

// check status with
assert response.status() == 200;

// parse body (a byte[]) with:
parseIntoJson(response.body())

// or use the helper method to read body into a map:
Map<String, Map<String, Object>> info = response.bodyAsMap();
```

Add query properties variables: 
```java
HttpResponse response = client.buildRequest("POST", "/ksql")
    .payload("ksql", "CREATE STREAM FOO AS CONCAT(A, `wow;`) FROM `BAR`;")
    .propertiesKey("streamsProperties")
    .property("auto.offset.reset", "earliest")
    .send()
    .get();
assert response.status() == 200;
```

Or build the entire payload manually: 
```java
HttpResponse response = client.buildRequest("POST", "/ksql")
    .payload("ksql", "CREATE STREAM FOO AS CONCAT(A, `wow;`) FROM `BAR`;")
    .payload("streamsProperties", Collections.singletonMap("auto.offset.reset", "earliest"))
    .send()
    .get();
assert response.status() == 200;
```

> The `send()` method will add authentication headers as specified in [ClientOptions](api/io/confluent/ksql/api/client/ClientOptions.html).

Reviewer: ```suggestion
The `send()` method adds authentication headers as specified in [ClientOptions](api/io/confluent/ksql/api/client/ClientOptions.html).
```
